pkg/cmd/openshift-network-controller/cmd.go

```diff
@@ -0,0 +1,166 @@
+		Run: func(c *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Validate())
+
+			serviceability.StartProfiler()
+
+			if err := options.StartNetworkController(); err != nil {
+				if kerrors.IsInvalid(err) {
+					if details := err.(*kerrors.StatusError).ErrStatus.Details; details != nil {
+						fmt.Fprintf(errout, "Invalid %s %s\n", details.Kind, details.Name)
+						for _, cause := range details.Causes {
+							fmt.Fprintf(errout, "  %s: %s\n", cause.Field, cause.Message)
+						}
+						os.Exit(255)
+					}
+				}
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	// This command only supports reading from config
+	flags.StringVar(&options.ConfigFilePath, "config", options.ConfigFilePath, "Location of the master configuration file to run from.")
+	cmd.MarkFlagFilename("config", "yaml", "yml")
+	cmd.MarkFlagRequired("config")
+
+	return cmd
+}
+
+func (o *OpenShiftNetworkController) Validate() error {
+	if len(o.ConfigFilePath) == 0 {
+		return errors.New("--config is required for this command")
+	}
+
+	return nil
+}
+
+// StartNetworkController calls RunNetworkController and then waits forever
+func (o *OpenShiftNetworkController) StartNetworkController() error {
+	if err := o.RunNetworkController(); err != nil {
+		return err
+	}
+
+	go daemon.SdNotify(false, "READY=1")
+	select {}
+}
+
+// RunNetworkController takes the options and starts the network controller
+func (o *OpenShiftNetworkController) RunNetworkController() error {
+	// try to decode into our new types first.  right now there is no validation, no file path resolution.  this unsticks the operator to start.
+	// TODO add those things
+	configContent, err := ioutil.ReadFile(o.ConfigFilePath)
+	if err != nil {
+		return err
+	}
+	scheme := runtime.NewScheme()
+	utilruntime.Must(openshiftcontrolplanev1.Install(scheme))
+	codecs := serializer.NewCodecFactory(scheme)
+	obj, err := runtime.Decode(codecs.UniversalDecoder(openshiftcontrolplanev1.GroupVersion, configv1.GroupVersion), configContent)
+	if err == nil {
+		// Resolve relative to CWD
+		absoluteConfigFile, err := api.MakeAbs(o.ConfigFilePath, "")
+		if err != nil {
+			return err
+		}
+		configFileLocation := path.Dir(absoluteConfigFile)
+
+		config := obj.(*openshiftcontrolplanev1.OpenShiftControllerManagerConfig)
+		/// this isn't allowed to be nil when by itself.
+		// TODO remove this when the old path is gone.
+		if config.ServingInfo == nil {
+			config.ServingInfo = &configv1.HTTPServingInfo{}
+		}
+		if err := helpers.ResolvePaths(configconversion.GetOpenShiftControllerConfigFileReferences(config), configFileLocation); err != nil {
+			return err
+		}
+		configdefault.SetRecommendedOpenShiftControllerConfigDefaults(config)
+
+		clientConfig, err := helpers.GetKubeClientConfig(config.KubeClientConfig)
+		if err != nil {
+			return err
+		}
+		return RunOpenShiftNetworkController(config, clientConfig)
+	}
+
+	masterConfig, err := configapilatest.ReadAndResolveMasterConfig(o.ConfigFilePath)
```

Comment:
Are you using this mode?  You need to remove it.  We are going to try to delete the entire MasterConfig when we land a rebase because nothing will be using it.  You don't want to be the last person using it.

So remove it now or a high severity bugzilla.  Our time horizon is 11/1 for removal.  @openshift/sig-master

Reply:
We don't use this in the operator-based install; I don't know if can be removed entirely.

Reply:
I just copied all that code from openshift-controller-manager. We can generate/parse the config file in whatever format we need to, I just copied the existing code so we could use the existing config for now.

Reply:
In that case, please drop the generate the config as you need fit.

Reply:
> We don't use this in the operator-based install; I don't know if can be removed entirely.

When the rebase lands, the operator-based install will be all we have.  Hopefully d-in-d isn't using this.